Add authenticated check for `Follows you`

jm90m opened the pull request:
Fixes #1418

Changes:
- add authenticated check for `Follows you`

Sekhmet commented:
I can't reproduce this problem. I'm not logged in and I can't make "Follows you" to display in any way (on production and staging as well).

bonustrack commented:
Did you tried on mobile @Sekhmet ?

Sekhmet commented:
Doesn't happen for me on mobile as well. Should I visit user profile page directly or by navigating there using links? Neither causes problem for me.

jm90m commented:
@Sekhmet Its happening for me right now on staging
<img width="917" alt="screen shot 2018-02-02 at 4 44 20 am" src="https://user-images.githubusercontent.com/5067716/35726712-cfffc3d4-07d3-11e8-9363-5ae3d3a9a31e.png">

it happens when you directly go to the page sometimes

Sekhmet commented on src/client/components/UserHeader.js:
```diff
@@ -103,7 +104,8 @@ const UserHeader = ({
           </div>
-          {isFollowing &&
+          {authenticated &&
+            isFollowing &&
```
I think `isFollowing` should take whether user is logged in or not in account. Can we update how `isFollowing` is calculated, so `isFollowing` really means that user is following me.

jm90m commented:
@Sekhmet good idea, just updated

Sekhmet commented:
As I can't reproduce I'm just guessing, but it wouldn't be better to set `isFollowing` to `false` by default and never check if user is following you, if you are not authenticated?

jm90m commented:
So this happens when you login and logout, and move around, looks like @bonustrack is experiencing it too on mobile - https://github.com/busyorg/busy/issues/1456

I updated it, I forgot why I was using a null check there, but could've been with something in the past, that is now fixed. Refactored it.
